src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java

```diff
@@ -110,7 +110,7 @@ public CoreCaseDataService(
     }
 
     @LogExecutionTime
-    public Claim createNewCase(User user, Claim claim) {
+    public Claim createNewCase(User user, Claim claim, CaseEvent caseEvent) {
```

Comment:
I don't think we should be re-using this method as it is:

-  It is setting the reference number of the claim, which we don't want to set here as per our callback approach.
- We are passing a `CaseEvent` object which can be anything, and we are interested only one/two of those, rest should be invalid arguments. 
1.